beacon-chain/rpc/eth/beacon/state.go

```diff
@@ -124,13 +138,18 @@ func (bs *Server) GetFinalityCheckpoints(ctx context.Context, req *ethpb.StateRe
 		}
 		return nil, status.Errorf(codes.Internal, "Could not get state: %v", err)
 	}
+	isOptimistic, err := helpers.IsOptimistic(ctx, st, bs.HeadFetcher)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not check if slot's block is optimistic: %v", err)
+	}
 
 	return &ethpb.StateFinalityCheckpointResponse{
 		Data: &ethpb.StateFinalityCheckpointResponse_StateFinalityCheckpoint{
 			PreviousJustified: checkpoint(st.PreviousJustifiedCheckpoint()),
 			CurrentJustified:  checkpoint(st.CurrentJustifiedCheckpoint()),
 			Finalized:         checkpoint(st.FinalizedCheckpoint()),
 		},
+		ExecutionOptimistic: isOptimistic,
```

Comment:
I see we are returning optimistic status in lots of requests. I do not see the specs being so comprehensive: in particular I only see mentions of requests for blocks and for heads here https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md, are all these changes necessary?

Reply:
resolved offline